### ISSUE TITLE
pyGHDL: 

### ISSUE THREAD

**Issue by BenWiederhake** [ghdl/ghdl#2779]
**Description**
pyVHDLModel provides a representation for variable assignments: https://vhdl.github.io/pyVHDLModel/pyVHDLModel/pyVHDLModel.Sequential.html#pyVHDLModel.Sequential.SequentialVariableAssignment
However, pyGHDL simply drops variable assignments, leaving not even an indication *where* the AST is incomplete: https://github.com/ghdl/ghdl/blob/9d905ffcdc0663dafbeee3d5d3fb49dcc2566b63/pyGHDL/dom/_Translate.py#L878

**Expected behaviour**
- When using pyGHDL to parse a file, variable assignment is supported, making the output `[NOT IMPLEMENTED] Variable assignment (label: 'None') at line 76` unnecessary.
- When using pyGHDL to parse a file, unsupported statements are somehow at least recoverable.

**How to reproduce?**

In a file `foo.vhd`:
```vhd :file: foo.vhd
entity foo is end entity;
architecture arch of foo is
begin
	process(all)
		variable bar : natural;
	begin
		bar := 42;
	end process;
end architecture;
```

Then run file `mre_pyghdl.py`:
```python3
#!/usr/bin/env python3
from pathlib import Path
import pyGHDL
from pyGHDL.dom.NonStandard import Design, Document
sourceFile = Path("foo.vhd")
design = Design()
library = design.GetLibrary("lib")
document = Document(sourceFile)
design.AddDocument(document, library)
```

**Context**
Please, provide the following information:

- OS: Debian Testing+Sid, using ghdl-gcc and libghdl-4-1-0
- Origin:
  - [X] Package manager: `4.1.0+dfsg-2+b1`
  - [ ] Released binaries: `tarball_url`
  - [ ] Built from sources: `commit SHA`

**Comment by Paebbels** [comment]
Yes, not all of GHDL's IIR is currently translated to `pyVHDLModel`. All spots are usually identified by print messages.

> However, pyGHDL simply drops variable assignments, leaving not even an indication where the AST is incomplete:

Thus, an incomplete "AST" is indicated by the warning message.

> * When using pyGHDL to parse a file, variable assignment is supported, making the output [NOT IMPLEMENTED] Variable assignment (label: 'None') at line 76 unnecessary.
> * When using pyGHDL to parse a file, unsupported statements are somehow at least recoverable.

So, when variable assignments get translated into the CodeDOM, this message will disappear. (contributions are of cause welcome). The parsing should be recovering, because it just ignores these assignments.

Translating assignments it not easy (1h job), because it requires:
* operators
* expressions
* function calls
* type conversions
* qualified expressions
* names (selected name, attribute name, indexed name, sliced name, ...)
* aggregates
* ...

**Comment by Paebbels** [comment]
This will be added in the next days.
